browser/resources/settings/brave_leo_assistant_page/model_config_ui.ts

```diff
@@ -0,0 +1,168 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import 'chrome://resources/cr_elements/cr_button/cr_button.js'
+import 'chrome://resources/cr_elements/icons.html.js'
+
+import { PrefsMixin } from '/shared/settings/prefs/prefs_mixin.js'
+import { I18nMixin } from 'chrome://resources/cr_elements/i18n_mixin.js'
+import { PolymerElement } from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js'
+import * as mojom from './brave_leo_assistant_browser_proxy.js'
+import { BaseMixin } from '../base_mixin.js'
+import { getTemplate } from './model_config_ui.html.js'
+
+const ModelConfigUIBase = PrefsMixin(I18nMixin(BaseMixin(PolymerElement)))
+
+class ModelConfigUI extends ModelConfigUIBase {
+  static get is() {
+    return 'model-config-ui'
+  }
+
+  static get template() {
+    return getTemplate()
+  }
+
+  static get properties() {
+    return {
+      label: {
+        type: String
+      },
+      modelRequestName: {
+        type: String
+      },
+      endpointUrl: {
+        type: String
+      },
+      apiKey: {
+        type: String,
+        value: ''
+      },
+      isUrlInvalid: {
+        type: Boolean,
+        value: false
+      },
+      modelItem: {
+        type: Object,
+        value: null,
+        notify: true,
+        observer: 'onModelItemChange_'
+      },
+      isEditing_: {
+        type: Boolean,
+        value: false,
+        readOnly: true,
+        computed: 'computeIsEditing_(modelItem)'
+      },
+      buttonLabel_: {
+        type: String,
+        computed: 'computeButtonLabel_(isEditing_)'
+      }
+    }
+  }
+
+  label: string
+  modelRequestName: string
+  endpointUrl: string
+  apiKey: string
+  modelItem: mojom.Model | null
+  isEditing_: boolean
+  isUrlInvalid: boolean
+
+  override ready() {
+    super.ready()
+  }
+
+  handleClick_() {
+    if (!this.saveEnabled_()) {
+      return
+    }
+
+    // We need to check if the URL is valid because sending bad URL will cause
+    // renderer to crash. This is mainly due to mojo IPC not being able to
+    // handle bad URLs properly for mojomUrl type
+    try {
+      new URL(this.endpointUrl)
+    } catch (error) {
+      this.isUrlInvalid = true
+      // @ts-expect-error
+      this.$['endpoint-input'].errorMessage = this.i18n(
+        'braveLeoAssistantEndpointInvalidError'
+      )
```

Comment:
Why do we need to set this when the `errorMessage` attribute is already set to this value in the html?

Reply:
added